lib/capybara-screenshot/s3_saver.rb

```diff
@@ -17,13 +18,13 @@ def self.new_with_configuration(saver, configuration)
         }
 
         s3_client_credentials = default_s3_client_credentials.merge(
-          configuration.fetch(:s3_client_credentials)
+          configuration.delete(:s3_client_credentials)
         )
 
         s3_client = Aws::S3::Client.new(s3_client_credentials)
-        bucket_name = configuration.fetch(:bucket_name)
+        bucket_name = configuration.delete(:bucket_name)
 
-        new(saver, s3_client, bucket_name)
+        new(saver, s3_client, bucket_name, **configuration)
```

Comment:
If you change to `options` for consistency, then you just pass in configuration?